community/kernel/src/main/java/org/neo4j/kernel/monitoring/Monitors.java

```diff
@@ -71,155 +74,81 @@ public Monitors( LogProvider logProvider )
         this.log = logProvider.getLog( Monitors.class );
     }
 
-    private static final AtomicBoolean FALSE = new AtomicBoolean( false );
-
-    // Concurrency: Mutation of these data structures is always guarded by the monitor lock on this Monitors instance,
-    // while look-ups and reads are performed concurrently. The methodMonitorListeners lists (the map values) are
-    // read concurrently by the proxies, while changing the listener set always produce new lists that atomically
-    // replace the ones already in the methodMonitorListeners map.
-
-    /** Monitor interface method -> Listeners */
-    private final Map<Method, List<MonitorListenerInvocationHandler>> methodMonitorListeners = new ConcurrentHashMap<>();
-
-    /**
-     * Monitor interface -> Has Listeners?
-     * Used to determine if recalculation of listeners is needed
-     */
-    private final Map<Class<?>,AtomicBoolean> monitoredInterfaces = new ConcurrentHashMap<>();
-
-    /**
-     * Listener predicate -> Listener
-     * Used to add listeners to monitors that are added after the listener
-     */
-    private final Map<Predicate<Method>, MonitorListenerInvocationHandler> monitorListeners = new ConcurrentHashMap<>();
-
     public synchronized <T> T newMonitor( Class<T> monitorClass, Class<?> owningClass, String... tags )
     {
-        Iterable<String> tagIer = append( owningClass.getName(), Iterables.iterable( tags ) );
-        String[] tagArray = asArray( String.class, tagIer );
-        return newMonitor( monitorClass, tagArray );
+        String[] monitorTags = ArrayUtil.concat( tags, owningClass.getName() );
+        return newMonitor( monitorClass, monitorTags );
     }
 
     public synchronized <T> T newMonitor( Class<T> monitorClass, String... tags )
     {
-        if ( !monitoredInterfaces.containsKey( monitorClass ) )
-        {
-            monitoredInterfaces.put( monitorClass, new AtomicBoolean( false ) );
-
-            for ( Method method : monitorClass.getMethods() )
-            {
-                recalculateMethodListeners( method );
-            }
-        }
-
+        requireInterface( monitorClass );
         ClassLoader classLoader = monitorClass.getClassLoader();
         MonitorInvocationHandler monitorInvocationHandler = new MonitorInvocationHandler( tags );
         return monitorClass.cast( Proxy.newProxyInstance( classLoader, new Class<?>[]{monitorClass}, monitorInvocationHandler ) );
     }
```

Comment:
I don't see why the `newMonitor` methods need to be synchronised.

Reply:
yes correct, now when the whole set of maps related magic is gone it should not be synch